olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,170 @@
+            ),
+            "batch_size": PassConfigParam(
+                type_=int,
+                default_value=1,
+                description="Batch size for quantization. Default value is 1.",
+            ),
+            "seed": PassConfigParam(
+                type_=int,
+                default_value=0,
+                description="Random seed for sampling calibration dataset. Default value is 0.",
+            ),
+            "damp_percent": PassConfigParam(
+                type_=float,
+                default_value=0.01,
+                description="Damping factor for quantization. Default value is 0.01.",
+            ),
+            "static_groups": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use static groups for quantization. Default value is False.",
+            ),
+            "true_sequential": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use true sequential for quantization. Default value is False.",
+            ),
+            "desc_act": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use descriptive activation for quantization. Default value is False.",
+            ),
+            "sym": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Symmetric quantization. Default value is False.",
+            ),
+            "gpu": PassConfigParam(
+                type_=bool,
+                default_value=True,
+                description="Use GPU for quantization. Default value is True.",
+            ),
+            "export_optimization": PassConfigParam(
+                type_=str,
+                default_value=None,
+                description="Export optimization level. Default value is None.",
+            ),
+        }
+
+    @torch.no_grad()
+    def _run_for_config(
+        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> ONNXModelHandler:
+        from onnxruntime import __version__ as ort_version
+        from optimum.exporters.onnx import onnx_export_from_model
+        from optimum.version import __version__ as optimum_version
+        from transformers import AutoModelForCausalLM, AutoTokenizer, GPTQConfig
+
+        from olive.passes.onnx.gptq_utils import QuantLinearORT
+
+        if version.parse(optimum_version) < version.parse("1.17.0"):
+            raise ValueError("Please use optimum>=1.17.0 for gptq quantization")
+        if version.parse(ort_version) < version.parse("1.17.0"):
+            raise ValueError("Please use onnxruntime-gpu>=1.17.0 for gptq quantization")
+        if (
+            self.accelerator_spec.accelerator_type != Device.GPU
+            or self.accelerator_spec.execution_provider != "CUDAExecutionProvider"
```

Comment:
do we need support tensorRT?

Reply:
I think it's not supported for now because we use our matmulnbits kernel in ort.